### PR TITLE
refactor: Create 'developer' folder

### PR DESCRIPTION
**Pull request by Roming22**
This commit is part of a larger set of commits with the goal of restructuring the repository.

Signed-off-by: Romain Arnaud <rarnaud@redhat.com>

### CODE REVIEW

**Comment by Roming22** [comment]
@xinredhat Tests should pass once https://github.com/openshift/release/pull/33403 is merged.

File: images/kcp-upgrade/upgrade.sh [hunk 1]
```diff
@@ -23,7 +23,7 @@ SCRIPT_DIR="$(
   pwd
 )"
 
-CONFIG="$(dirname "$(dirname "$SCRIPT_DIR")")/config/config.yaml"
+CONFIG="$(dirname "$(dirname "$SCRIPT_DIR")")/ckcp/config.yaml"
```
Comment (ramessesii2) [inline images/kcp-upgrade/upgrade.sh:26]: It's under `/developer` now.
```suggestion
CONFIG="$(dirname "$(dirname "$SCRIPT_DIR")")/developer/ckcp/config.yaml"
```

**Review by ramessesii2 — COMMENTED** [review]
Overall, looks good to me. We'll need to rebase so that the `.github/workflows` has latest changes.

File: images/kcp-upgrade/upgrade.sh (hunk 1 above)
Comment (Roming22) [reply images/kcp-upgrade/upgrade.sh]: Done

File: images/kcp-upgrade/upgrade.sh [hunk 2]
```diff
@@ -38,12 +38,12 @@ if [ "$current_kcp_version" != "$latest_kcp_version" ]; then
   sed -i "s,$current_kcp_version,$latest_kcp_version,g" .github/workflows/build-push-images.yaml
   sed -i "s,$current_kcp_version,$latest_kcp_version,g" .github/workflows/local-dev-ci.yaml
   sed -i "s,$current_kcp_version,$latest_kcp_version,g" DEPENDENCIES.md
-  sed -i "s,$current_kcp_version,$latest_kcp_version,g" ckcp/openshift/overlays/dev/kustomization.yaml
+  sed -i "s,$current_kcp_version,$latest_kcp_version,g" developer/ckcp/openshift/overlays/dev/kustomization.yaml
   sed -i "s,$current_kcp_version,$latest_kcp_version,g" docs/kcp-registration.md
```
Comment (bnallapeta) [inline images/kcp-upgrade/upgrade.sh:42]: path needs to be updated to architect/docs/kcp-registration.md

File: developer/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Documentation for Developers
```
Comment (bnallapeta) [inline developer/docs/README.md:1]: Not necessarily a change in this PR, but I would like to improve this content in a future PR to explain we have the devenv env and local env and ckcp env for developers to test Pipeline Service.

**Review by bnallapeta — APPROVED** [review]
Minor change. LGTM.

**Review by Mo3m3n — COMMENTED** [review]
- shouldn't the ckcp folder disappear from root ? currently it is still there with "manifests" and "test.sh" as content
- You will need to fix the commit subject and replace architect by developer

File: images/kcp-upgrade/upgrade.sh (hunk 2 above)
Comment (Roming22) [reply images/kcp-upgrade/upgrade.sh]: Done (but it's going to move again soon :wink: )

**Pull request merged**